### ISSUE TITLE
Model no attribute 'id' after creating

### ISSUE THREAD

**Issue by resmo** [MasoniteFramework/orm#390]
**Describe the bug**

Storing a new model does not allow to access the `id` attribute. Accessing in code after `save()` results in a 
`class model 'Category' has no attribute id`

```python
category.save()
request.session.flash('success', f'Saved: {category.id}')
return request.redirect_to('admin.category.table')
```

**To Reproduce**

create a simple store function:

```python
from app.Category import Category

class CategoryController(Controller):

    def store(self, request: Request):
        category = Category()
        category.name = request.input('name')
        category.slug = request.input('slug')
        category.description = request.input('description')
        category.sorting = request.input('sorting')
        category.save()
        return category
```

**Expected behavior**
What do you believe should be happening?
```json
{
  "id": 1,
  "name": "test",
  "slug": "test",
  "description": "test",
  "sorting": "1"
}
```

**Actual behavior**
```json
{
  "name": "test",
  "slug": "test",
  "description": "test",
  "sorting": "1"
}
```

**Screenshots or code snippets**
**What database are you using?**
 - Type: SQLite 2.8.17
 - Masonite ORM:  masonite-orm==1.0.27

**Comment by Marlysson** [comment]
@resmo Which python version are you using to come with this sqlite version?

**Comment by josephmancuso** [comment]
@Marlysson that's a really good catch. That SQLite version is from 2005 ...

**Comment by Marlysson** [comment]
@resmo, I'm trying reproduce that here and I've notice some litte problem. 
You're just saving the entity, but the inserted result is forgotten, you need to assign the result of the `.save()` operation back to category variable.

```python
 def store(self, request: Request):
    category = Category()
    # operations
    category = category.save() # Assign back to variable.
    return category
```

With this you will be able to get the id of model.

**Comment by resmo** [comment]
@Marlysson I confirm, assigning back to a var works.

However, this assignment is totally not usual for an ORM, especially not how orator works https://orator-orm.com/docs/0.9/orm.html#insert-update-and-delete

**Comment by josephmancuso** [comment]
@Marlysson @resmo is correct, His code should be working as he is intending it to work. I think the issue is we are just missing a fill.

```python
            if self.is_loaded():
                result = builder.update(self.__dirty_attributes__)
            else:
                result = self.create(self.__dirty_attributes__, query=query)
            self.observe_events(self, "saved")
            self.fill(result) # <--- this is missing here. Assigns the result to the current model
            return result
```

**Comment by josephmancuso** [comment]
@resmo releasing 1.0.29 now to fix this. Thanks for the report!

**Comment by marcoscleison** [comment]
Dear developers, thank you for all the effort to have this library working. 
I am experimenting the same bug today. 
I installed via pip:
```bash
pip install masonite-orm
```
The version installed is:
masonite-orm==3.0.0

This is a little different because I do not see it in release. But its ok.
My python is 3.11 in  Linux/Ubunutu
I downgraded to masonite-orm==2.23.1 and it worked well without the issue.
Thank you for the attention.
